plugins/foreman_kubevirt/0.x/index.md

```diff
@@ -0,0 +1,311 @@
+
+## 3.2 Red Hat, CentOS, Fedora, Scientific Linux (rpm)
+
+Set up the repositories as explained in the [core documentation](https://theforeman.org/manuals/1.21/index.html#3.3.1RPMPackages) to install the package with:
+
+    $ yum install tfm-rubygem-foreman_kubevirt
+
+## 3.3 Debian, Ubuntu (deb)
+
+Add the repository sources as described in the [core documentation](https://theforeman.org/manuals/1.21/index.html#3.3.3DebianPackages) and install the package:
+
+    $ apt-get install ruby-foreman-kubevirt
+
+## 3.4 Bundle (gem)
+
+This method will **only** work on Foreman deployments installed from source. Please refrain from making these changes if you have installed Foreman via packages.
+
+Add the following to bundler.d/Gemfile.local.rb in your Foreman installation directory (/usr/share/foreman by default)
+
+    $ gem 'foreman_kubevirt'
+
+Then run `bundle install` from the same directory
+
+--------------
+
+To verify that the installation was successful, go to Foreman, top bar **Administer > About** and check 'foreman_kubevirt' shows up in the **System Status** menu under the Plugins tab.
+
+# 4. Usage
+
+Go to **Infrastructure > Compute Resources** and click on **New Compute Resource**.
+Choose the **KubeVirt provider**, and fill in all the fields.
+
+Here is a short description of some of the fields:
+* *Namespace* - the virtual cluster on kubernetes to which the user has permissions as cluster-admin.
+* *Token* - a bearer token authentication for HTTP(s) calls.
+* *X509 Certification Authorities* - enables client certificate authentication for API server calls.
+
+## 4.1 Setting Compute Resource *Token* and *X509 CA* values
+
+### Kubernetes
+#### *Token*:
+
+Either list the secrets and pick the one that contains the relevant token, or select a service account:
+
+List of secrets that contain the tokens and set secret name instead of *YOUR_SECRET*:
+```
+# kubectl get secrets
+# kubectl get secrets YOUR_SECRET -o jsonpath='{.data.token}' | base64 -d | xargs
+```
+
+Or obtain token for a service account named 'foreman-account':
+```
+# KUBE_SECRET=`kubectl get sa foreman-account -o jsonpath='{.secrets[0].name}'`
+# kubectl get secrets $KUBE_SECRET -o jsonpath='{.data.token}' | base64 -d | xargs
+```
+
+#### *X509 CA*:
+
+Taken from Kubernetes admin config file:
+```
+# cat /etc/kubernetes/admin.conf | grep certificate-authority-data: | cut -d: -f2 | tr -d " " | base64 -d
+```
+
+Or by retrieving from the secret, via the service account (in this example assuming its name is *foreman-account*):
+```
+# KUBE_SECRET=`kubectl get sa foreman-account -o jsonpath='{.secrets[0].name}'`
+# kubectl get secret $KUBE_SECRET  -o jsonpath='{.data.ca\.crt}' | base64 -d
+```
+
+### OpenShift
+#### *Token*:
+
+Create a privileged account named *my-account*:
+```
+# oc create -f https://raw.githubusercontent.com/ManageIQ/manageiq-providers-kubevirt/master/manifests/account-openshift.yml
```

Comment:
should this be hosted on this repo (or on kubevirt repo perhaps?)

Reply:
I'd prefer to have it here as instead of having this link, so I'll update this section with:

```

cat <<EOF | kubectl create -f -
apiVersion: v1
kind: ServiceAccount
metadata:
  name: foreman-account
  namespace: default
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: foreman-cluster-admin
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: foreman-account
  namespace: default
EOF
```